src/main/java/de/adesso/kicker/notification/NotificationService.java

```diff
@@ -103,10 +95,10 @@ public void checkWrongReceiver(Notification notification) {
         }
     }
 
-    public void checkNotificationExists(long notificationId) {
+    public void checkNotificationExists(Notification notification) {
 
-        if (!notificationRepository.existsById(notificationId)) {
-            throw new NotificationNotExistingException(notificationId);
+        if (notification == null) {
```

Comment:
Looks like I didn't properly describe what I meant. That was the solution I originally suggested. What I meant now, though possibly a bit hard to understand, is that the `CrudRepository` provides a method name `findById(String id)` with returns an `Optional`. `Optional` has a method called `orElseThrow(Exception.class)` which either returns the object inside the `Optional` or if `null` throw the specified exception. This is, in my opinion, a nicer approach as we take advantage of the provided API instead of doing stuff by hand.